software/profiling/TAU-2024.rst

```diff
@@ -0,0 +1,898 @@
+| -optPdtF95Reset=""         |    Reset options for Fortran parser in PDT                    		    |
++----------------------------+------------------------------------------------------------------------------+
+| -optPdtCOpts=""            |    Options for C parser in PDT                                               |
++----------------------------+------------------------------------------------------------------------------+
+| -optPdtCXXOpts=""          |    Options for C++ parser in PDT                                             |
++----------------------------+------------------------------------------------------------------------------+
+
+
+MiniWeather Example Application
+================================
+
+Getting the source code
+-----------------------
+
+- Connect to Summit and navigate to your project space
+- For the following examples, we'll use the MiniWeather application:
+  https://github.com/mrnorman/miniWeather
+
+.. code::
+
+	$ git clone https://github.com/mrnorman/miniWeather.git
+
+
+Compile the application
+------------------------
+
+- We'll use the PGI compiler; this application supports serial, MPI, MPI+OpenMP,
+  and MPI+OpenACC
+
+.. code::
+
+	$ module load pgi
+	$ module load parallel-netcdf
+
+- Different compilations for Serial, MPI, MPI+OpenMP, and MPI+OpenACC:
+
+.. code::
+
+	$ module load cmake
+	$ cd miniWeather/c/build
+        $ ./cmake_summit_pgi.sh
+	$ make serial
+	$ make mpi
+	$ make openmp
+	$ make openacc
+
+
+Below, we'll look at using TAU to profile each case.
+
+
+Modifications
+-------------
+
+- Edit the cmake_summit_pgi.sh and replace ``mpic++`` with ``tau_cxx.sh``. This applies
+  only for the non-GPU versions.
+- TAU works with special TAU makefiles to declare what programming models are
+  expected from the application:
+
+  - The available makefiles are located inside TAU installation:
+
+.. code::
+
+        $ module show tau
+        ---------------------------------------------------------------
+           /sw/summit/modulefiles/core/tau/2.28.1:
+        ---------------------------------------------------------------
+        whatis("TAU 2.28.1 github ")
+        setenv("TAU_DIR","/sw/summit/tau/tau2/ibm64linux")
+        prepend_path("PATH","/sw/summit/tau/tau2/ibm64linux/bin")
+        help([[https://www.olcf.ornl.gov/software_package/tau
+        ]])
+
+
+The available Makefiles are named per-compiler and are located in:
+
+.. code::
+
+        $ ls ${TAU_DIR}/lib/Makefile.tau-pgi*
+        /sw/summit/tau/tau2/ibm64linux/lib/Makefile.tau-pgi-papi-mpi-cupti-pdt-openmp-pgi
+        /sw/summit/tau/tau2/ibm64linux/lib/Makefile.tau-pgi-papi-mpi-cupti-pdt-pgi
+        /sw/summit/tau/tau2/ibm64linux/lib/Makefile.tau-pgi-papi-pdt-pgi
+        /sw/summit/tau/tau2/ibm64linux/lib/Makefile.tau-pgi_memory_manager-papi-mpi-cupti-pdt-pgi
+
+- To list all TAU makefiles:
+
+.. code::
+
+        $ ls ${TAU_DIR}/lib/Makefile.tau*
+
+
```

Comment:
As you mentioned, remove everything below as it is related to Summit. Future effort to redesign miniweather instructions to be compatible with Frontier.